Contracts generator (#1575)
0xProject/0x-monorepo: packages/contracts-gen/README.md

# Contracts Gen

This package allows you to generate boilerplate TypeScript code and configs for smart contracts packages.

## Installation

`yarn add -g @0x/contracts-gen`

## Usage

Run it from within your smart contracts packages.

```bash
contracts-gen
```

You should run this tool after each time you move your contracts around to regenerate boilerplate code and configs.

## What can it generate

This tool does the following:

> -   Reads your `compiler.json`. Specifically list of smart contracts.

Reviewer: ```suggestion
-   Reads your `compiler.json`. Specifically the list of smart contracts.
```

-   Creates `wrapper.ts` file which exports all contract wrappers.
-   Creates `artifacts.ts` file which exports all contract artifacts.
-   Generates list of JSON artifact files in `tsconfig.json`
-   Generates a glob for abi-gen in `package.json`

On top of that - if your `compiler.json` has contracts referenced just by name - it will resolve the name to relative path and put it there.

> It also sorts all the lists in it's output leading to smaller and cleaner diffs.

Reviewer: ```suggestion
It also sorts the list of smart contracs leading to smaller and cleaner diffs.
```

Author: Which list? It sorts multiple lists of different nature. One is the list of smart contract names. One is the list of imports for wrappers. One is the list of imports for artifacts. One is the artifacts JSON files list. One is the list of object keys for artifacts.

Reviewer: Ah, didn't catch that. Ignore.
